drivers/clock_control/clock_control_mcux_ccm_rev2.c

```diff
@@ -106,8 +106,28 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 	default:
 		return -EINVAL;
 	}
-
+#ifdef CONFIG_SOC_MIMX93_A55
+	*rate = CLOCK_GetIpFreq(clock_root);
+#else
 	*rate = CLOCK_GetRootClockFreq(clock_root);
+#endif
+	return 0;
+}
+
+static int mcux_ccm_init(const struct device *dev)
+{
+#ifdef CONFIG_SOC_MIMX93_A55
+	g_clockSourceFreq[kCLOCK_Osc24M] 			= 24000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1]			= 4000000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd0]		= 1000000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd0Div2]	= 500000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd1]		= 800000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd1Div2]	= 400000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd2]		= 625000000U;
+	g_clockSourceFreq[kCLOCK_SysPll1Pfd2Div2]	= 312500000U;
+	g_clockSourceFreq[kCLOCK_AudioPll1]			= 393216000U;
+	g_clockSourceFreq[kCLOCK_AudioPll1Out]		= 393216000U;
```

Comment:
As we discussed, these clock source frequency need to be initialized in hal driver, we already have such fix, so port the fix to hal_nxp firstly.

Reply:
ported. 
https://github.com/zephyrproject-rtos/hal_nxp/pull/286